packages/@lwc/template-compiler/src/__tests__/elseif.spec.ts

```diff
@@ -70,58 +70,60 @@ describe('lwc if/elseif/else directives', () => {
             expect(warnings.length).toBe(1);
             expect(warnings[0]).toMatchObject({
                 level: DiagnosticLevel.Error,
-                message: `LWC1158: 'lwc:else' directive must be used immediately after an element with 'lwc:if' or 'lwc:elseif'. No such element found.`,
+                message: `LWC1159: 'lwc:else' directive must be used immediately after an element with 'lwc:if' or 'lwc:elseif'. No such element found.`,
                 location: EXPECTED_LOCATION,
             });
         });
     });
 
     describe('invalid uses', () => {
-        it('multiple directives on single element - if and elseif', () => {
+        it('should throw an error when lwc:else has an expression value', () => {
             const { warnings } = parseTemplate(
-                `<template><template lwc:if={visible} lwc:elseif={elseif}>Conditional Text</template></template>`
+                `<template>
+                    <template lwc:if={visible}>Conditional Text</template>
+                    <template lwc:else={invalidCondition}>Invalid Text</template>
+                </template>`
             );
 
             expect(warnings.length).toBe(1);
-            expect(warnings[0]).toMatchObject({
-                level: DiagnosticLevel.Error,
-                message: `LWC1155: Invalid usage of 'lwc:if' and 'lwc:elseif' directives on the same element.`,
-                location: EXPECTED_LOCATION,
-            });
+            expect(warnings[0]).toMatchObject({});
         });
-        it('multiple directives on single element - if and else', () => {
+        it('should throw an error when lwc:else has a string value', () => {
             const { warnings } = parseTemplate(
-                `<template><template lwc:else lwc:if={visible}>Conditional Text</template></template>`
+                `<template>
+                    <template lwc:if={visible}>Conditional Text</template>
+                    <template lwc:else="invalid">Invalid Text</template>
+                </template>`
+            );
```

Comment:
Could we also add a test case for a mix of template and an HTML element?

```
<template>
    <template lwc:if={visible0}>
        <div lwc:if={visible}>if test</div>
        <template lwc:elseif={visible2}>else if test</template>
        <div lwc:else>else test</div>
    </template>
</template>
```

Reply:
Are you asking about adding a test case for that in general, or specifically as an "Invalid Case"?

For "invalid case", I don't think it's something we need to error on. Maybe it's clunky, but has its use:

```
<template>
    <template lwc:if={complex}>
        <div>
            <!-- some expanded set of elements -->
        </div>
        <div>
            <!-- more stuff -->
        </div>
    </template>
    <div lwc:else>Simple Text</div>
</template>
```

Also, for if:true and if:false, it doesn't matter whether they're template or HTML elements (since they're independent).

Do you see something that makes you want to error for that use case?

Reply:
Not for a "Invalid Case" sorry for the confusion (probably should've put this comment somewhere else), I meant just as a general test case so we can track that having a mix of template and HTML elements in the if, else-if, else chain should work.